### PR TITLE
Jakarta EE 9 transition

### PR DESCRIPTION
**Pull request by basil**
Downstream of #183. Do _not_ review. Merely filing this PR to have a record of a timestamped snapshot I just deployed. The timestamped snapshot is available here:

- https://repo.jenkins-ci.org/snapshots/org/jenkins-ci/winstone/5.22-SNAPSHOT/winstone-5.22-20211219.205002-3.jar
- https://repo.jenkins-ci.org/snapshots/org/jenkins-ci/winstone/5.22-SNAPSHOT/winstone-5.22-20211219.205002-3-sources.jar

### CODE REVIEW

File: src/main/java/winstone/Http2ConnectorFactory.java [hunk 1]
```diff
@@ -66,26 +66,28 @@ public Connector start( Map args, Server server ) throws IOException
...
+            //http2Connector.setHost( listenAddress );
+            //server.addConnector(http2Connector);
+            //server.setDumpAfterStart( Boolean.getBoolean( "dumpAfterStart" ) );
 
-            ALPN.debug = Boolean.getBoolean( "alpnDebug" );
+            //ALPN.debug = Boolean.getBoolean( "alpnDebug" );
 
-            return http2Connector;
+            //return http2Connector;
+            return null;
```
Comment (basil) [inline src/main/java/winstone/Http2ConnectorFactory.java:90]: This did not compile easily. Needs more investigation.

File: src/main/java/winstone/Launcher.java [hunk 2]
```diff
@@ -344,7 +343,8 @@ public static void main(String argv[]) throws IOException {
               }
           }
         }
-        Log.setLog(new JavaUtilLog());  // force java.util.logging for consistency & backward compatibility
+        // TODO needs triage
+        //Log.setLog(new JavaUtilLog());  // force java.util.logging for consistency & backward compatibility
```
Comment (basil) [inline src/main/java/winstone/Launcher.java:347]: Ditto.

File: src/main/java/winstone/ServerConnectorBuilder.java [hunk 3]
```diff
@@ -69,7 +69,9 @@ public ServerConnector build() {
         ServerConnector sc;
 
         if (sslContextFactory != null) {
-            sc = new ServerConnector(server, acceptors, selectors, sslContextFactory);
+            // TODO needs triage
+            //sc = new ServerConnector(server, acceptors, selectors, sslContextFactory);
+            throw new IllegalArgumentException("TODO needs triage");
```
Comment (basil) [inline src/main/java/winstone/ServerConnectorBuilder.java:74]: Ditto.

**Comment by olamy** [comment]
at least you could preserve history/contributions from people who did the job in another PR

**Comment by basil** [comment]
> at least you could preserve history/contributions from people who did the job in another PR

Please see the review description:

> Do _not_ review.

I am not sure why you are reviewing this PR when the description says not to review it. This PR is not intended to be merged. It is only intended for me to get an incremental build for my personal use. The PR that is intended to be merged is your PR, which attributes all credit to you (and rightly so!).

**Comment by joakime** [comment]
> I am not sure why you are reviewing this PR when the description says not to review it. This PR is not intended to be merged. It is only intended for me to get an incremental build for my personal use. The PR that is intended to be merged is your PR, which attributes all credit to you (and rightly so!).

Seems like a personal fork of the repo would be more appropriate way to use git and github for your personal reasons.

**Comment by narendrabtechcse** [comment]
Thank you for a pull request! Please check [this document](http://jenkins-ci.org/pull-request-greeting) for how the Jenkins project handles pull requests

**Comment by olamy** [comment]
would be better to base ee9/ee10 upgrade on Jetty 12 see https://github.com/jenkinsci/winstone/pull/341
note it's jdk17 required

**Pull request closed without merging**